middleware/2faMessageContent.js

```diff
@@ -0,0 +1,99 @@
+const hex = require('hexer');
+const nearAPI = require('near-api-js');
+const escapeHtml = require('escape-html');
+
+const fmtNear = (amount) => nearAPI.utils.format.formatNearAmount(amount, 4) + 'Ⓝ';
+
+const formatArgs = (args) => {
+    const argsBuffer = Buffer.from(args, 'base64');
+    try {
+        const jsonString = argsBuffer.toString('utf-8');
+        const json = JSON.parse(jsonString);
+        if (json.amount) json.amount = fmtNear(json.amount);
+        if (json.deposit) json.deposit = fmtNear(json.deposit);
+        return JSON.stringify(json);
+    } catch(e) {
+        // Cannot parse JSON, do hex dump
+        return hex(argsBuffer);
+    }
+};
+
+const formatAction = (receiver_id, { type, method_name, args, deposit, amount, public_key, permission }) => {
+    switch (type) {
+    case 'FunctionCall':
+        return escapeHtml(`Calling method: ${ method_name } in contract: ${ receiver_id } with amount ${ deposit ? fmtNear(deposit) : '0' } and with args ${formatArgs(args)}`);
+    case 'Transfer':
+        return escapeHtml(`Transferring ${ fmtNear(amount) } to: ${ receiver_id }`);
+    case 'Stake':
+        return escapeHtml(`Staking: ${ fmtNear(amount) } to validator: ${ receiver_id }`);
+    case 'AddKey':
+        if (permission) {
+            const { allowance, receiver_id, method_names } = permission;
+            const methodsMessage = method_names && method_names.length > 0 ? `${method_names.join(', ')} methods` : 'any method';
+            return escapeHtml(`Adding key ${ public_key } limited to call ${methodsMessage} on ${receiver_id} and spend up to ${fmtNear(allowance)} on gas`);
+        }
+        return escapeHtml(`Adding key ${ public_key } with FULL ACCESS to account`);
+    case 'DeleteKey':
+        return escapeHtml(`Deleting key ${ public_key }`);
+    }
+};
```

Comment:
The code in these methods was copied _as-is_ from `middleware/2fa.js`; no changes were made in this PR.